### PR TITLE
Add `QueryParser#missing_value` for handling missing values + tests.

### PR DESCRIPTION
**Pull request by ioquatix**
This is required for customizing how `Rack::QueryParser` handles missing values and make it easy for sub-classes to change this behaviour. Based on <https://github.com/rack/rack/pull/2038#issuecomment-1436060688>.

For the purpose of Rails:

```ruby
module ActionDispatch
  class Request
    # ...
    if Rack.release >= "3"
      class QueryParser < ::Rack::QueryParser
        def missing_value
          nil
        end
      end

      def query_parser
        QueryParser
      end
    end
```

A different design option would be to make this an argument, e.g. `QueryParser.new(missing_value: nil)`.

cc @matthewd @tenderlove

### CODE REVIEW

File: lib/rack/query_parser.rb [hunk 1]
```diff
@@ -207,8 +216,8 @@ def params_hash_has_key?(hash, key)
       true
     end
 
-    def unescape(s)
-      Utils.unescape(s)
+    def unescape(string, encoding = Encoding::UTF_8)
+      URI.decode_www_form_component(string, encoding)
```
Comment (jeremyevans) [inline lib/rack/query_parser.rb:220]: This seems unrelated to the change being made.  I'm not necessary opposed to this, but it shouldn't be part of this PR.  It can be a separate PR with a reason given for the change.  Ditto for the `uri` require at the top.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
I'm fine with this in principle. One inline comment.

File: lib/rack/query_parser.rb (hunk 1 above)
Comment (ioquatix) [reply lib/rack/query_parser.rb:220]: I couldn't make the separate tests pass without it.

Comment (jeremyevans) [reply lib/rack/query_parser.rb:220]: Fix for this is simple, submitted in #2055.  However, considering this was backported without any discussion, I'm not sure what we want to do here.
